server/src/main/java/org/elasticsearch/action/admin/indices/resolve/TransportResolveClusterAction.java

```diff
@@ -93,6 +118,15 @@ protected void doExecuteForked(Task task, ResolveClusterActionRequest request, A
         if (ccsCheckCompatibility) {
             checkCCSVersionCompatibility(request);
         }
+
+        // Fail the action if the specified timeout exceeds the value set in elasticsearch.yml.
+        TimeValue timeout = request.getTimeout();
+        if (timeout != null && timeout.getSeconds() > maxTimeoutInSeconds) {
+            throw new IllegalArgumentException(
```

Comment:
Rejecting requests like this is going to be super irritating to users. Imagine they have two clusters, one with a max timeout of 1s and the other with the default of 10s, and they want to get a response back in at most 5s. They're going to have to customize their requests based on which cluster they send it to.

Let's accept any timeout value. If ES returns an error sooner than their requested timeout then that's no big deal.